dashboard/app/models/levels/bubble_choice.rb

```diff
@@ -172,12 +172,24 @@ def summarize_sublevels(script_level: nil, user_id: nil, should_localize: false)
 
   # Returns the sublevel for a user that has the highest best_result.
   # @param [User]
-  # @return [Integer]
+  # @param [Script]
+  # @return [Level]
   def best_result_sublevel(user, script)
     ul = user.user_levels.where(level: sublevels, script: script).max_by(&:best_result)
     ul&.level
   end
 
+  def keep_working_sublevel(user, script)
+    # get latest feedback on sublevels where keepWorking is true
+    level_ids = sublevels.map(&:id)
+    latest_feedbacks = TeacherFeedback.get_latest_feedbacks_received(user.id, level_ids, script.id)
+
+    if latest_feedbacks.any?
+      keep_working_feedback = latest_feedbacks&.find {|feedback| feedback.review_state == TeacherFeedback::REVIEW_STATES.keepWorking}
```

Comment:
Not to complicate things here, but do we need to handle the case where more than one sublevel has a "keepWorking" state (but maybe different completion statuses)?

(I'm ok if we just take the first, but maybe include a comment that it's intentional.)

Reply:
The rule set by Amanda is, if any sublevel has a keepWorking status, the overall bubble has that status